internal/core/server/workspace.go

```diff
@@ -263,15 +270,32 @@ func (ws *Workspace) newController(ctx context.Context, desc api.ComponentDescri
 }
 
 // TODO: Use workspace-defined environments, rather than ambient unix environment.
-func (ws *Workspace) getEnvironment() map[string]string {
-	env := make(map[string]string)
-	for _, assign := range os.Environ() {
-		parts := strings.SplitN(assign, "=", 2)
-		key := parts[0]
-		val := parts[1]
-		env[key] = val
-	}
-	return env
+func (ws *Workspace) getEnvironment(ctx context.Context) (map[string]string, error) {
+	envPath, err := ws.resolveWorkspacePath(ctx, ".env")
+	if err != nil {
+		return nil, fmt.Errorf("could not resolve path to .env file: %w", err)
```

Comment:
the convention that the stdlib and core contributor examples tends to use is something more like `resolving path to .env file: %w`